rlclientlib/logger/file/file_logger.cc

```diff
@@ -0,0 +1,39 @@
+#include <sys/stat.h>
+#include "file_logger.h"
+#include "err_constants.h"
+#include "api_status.h"
+#include <fstream>
+namespace reinforcement_learning { namespace logger { namespace file {
+
+  file_logger::file_logger(const std::string& file_name, i_trace* trace)
+  : _file_name(file_name),
+  _trace(trace)
+  {}
+
+  file_logger::~file_logger()
+  {
+      _file.close();
+  }
+
+  int file_logger::init(api_status* status) {
```

Comment:
Can this object be changed to a factory method that returns a valid file logger and the constructor or factory method does the open. It will just help reduce the potential for unitialized objects.

Reply:
Since we have restriction on throw we construct and init() everywhere in the library.   This class is instantiated by a factory as are all other loggers.  But the factory itself is generic across many object types.  Does not seem appropriate to call init() from the factory.  live_model_impl ensures that init() will always be called.